script.js

```diff
@@ -1,4 +1,146 @@
+var attempts = 0;
+var wins = 0;
+var playerName = "";
+var gamemode = "normal";
+var currentWinner = "";
+
+// right now attempts & win count carries over for normal and reversed game, but not korean version
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (playerName == "") {
+    playerName = input;
+    return (
+      `Hi ${playerName}, please enter Scissors or Paper or Stone to start playing the conventional game!` +
+      "<br><br> Alternatively, enter 'reverse' anytime to play the reversed version of the game. <br><br> You can also enter 'korean' to play the Korean version of the game. <br><br> Enter 'normal' to revert back to the conventional game!"
+    );
+  }
+  input = input.toLowerCase();
+
+  // generating computer choice
+  var genRandomNumber = Math.floor(Math.random() * 3 + 1);
+  var computerChoice = "scissors";
+  if (genRandomNumber == 1) {
+    computerChoice = "paper";
+  }
+  if (genRandomNumber == 2) {
+    computerChoice = "stone";
+  }
```

Comment:
To make this code reusable, could have put this in a function named something like 
`getComputerChoice()` and return either scissors/paper/stone as the return value for this function